### PR TITLE
Add pkg-config support

### PR DESCRIPTION
**Pull request by carstene1ns**
Pkg-config is a general tool to aid in dependency resolution while building programs/libraries (more about pkg-config [here](https://www.freedesktop.org/wiki/Software/pkg-config/)). Adding support for it, will come in handy for applications using this library, when they use build systems like GNU autotools or query pkg-config for their dependent libraries directly.

This Pull Request adds a generated file containing various information about the library to the install process. As a safe-guard the availability of the tool itself is checked before.

*This has only been tested on GNU/Linux, but should work on other Unices and MinGW.*

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -220,6 +220,12 @@ IF(WIN32)
     ENDIF()
 ENDIF(WIN32)
 
+# pkg-config support
+find_package(PkgConfig)
+IF(NOT PKG_CONFIG_FOUND)
+    MESSAGE(STATUS "Will not provide a pkg-config file.")
+ENDIF(NOT PKG_CONFIG_FOUND)
+
```
Comment (psi29a) [inline CMakeLists.txt]: Shouldn't this be guarded to prevent it from being run on OSes without pkg-config support like Windows?

**Comment by psi29a** [comment]
Thank you for the PR!  My only concern is that it will break on Windows and other platforms without pkg-config support.

**Comment by sezero** [comment]
It won't 'break' windows, but checking pkg-config existence here is useless: we _don't need_ pkg-config to generate the pc file. if pkg-config doesn't exist, then the installed pc file will be useless on that setup and that is harmless.

**Comment by psi29a** [comment]
If it is a NOOP, then I see no problems with it. :)

**Pull request merged**